packages/node_modules/nav-frontend-stegindikator/src/stegindikator.tsx

```diff
@@ -0,0 +1,222 @@
+import * as React from 'react';
+import * as cn from 'classnames';
+
+import { omit } from 'nav-frontend-js-utils';
+
+import 'nav-frontend-stegindikator-style';
+
+const cls = (state) => cn('stegindikator', {
+    'stegindikator--kompakt': state.kompakt
+});
+
+const stegCls = (props) => cn('stegindikator__steg', {
+    'stegindikator__steg--labelled': props.visLabel
+});
+
+const innerCls = (props) => cn('stegindikator__steg-inner', {
+    'stegindikator__steg-inner--aktiv': props.aktiv,
+    'stegindikator__steg-inner--ferdig': props.ferdig,
+    'stegindikator__steg-inner--disabled': props.disabled,
+    'stegindikator__steg-inner--interaktiv': typeof props.onClick === 'function' && !props.aktiv
+});
+
+export interface StegindikatorStegProps {
+    label: string;
+    index: number;
+    visLabel?: boolean;
+    onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
+    aktiv?: boolean;
+    ferdig?: boolean;
+    disabled?: boolean;
+}
+
+export interface NavFrontendStegindikatorProps {
+    steg: StegindikatorStegProps[];
```

Comment:
Litt ref den diskusjonen som har vært nedentil angående `ferdigSteg`. 
`ferdig` og `aktiv` er allerede en del av `StegindikatorStegProps` så kan man ikke bare bruke disse for å evt. finne `aktivtSteg` og `ferdigSteg` om det er nødvendig å ha ett forhold til det i parent-komponenten?

Per i dag virker APIet til komponenten litt uklart for meg, det er kanskje fordi jeg bare sitter å ser på koden. 
Men om tar to steg tilbake (pun intended), så ville jeg foretrukket ett api alà:
```
<StegIndikator onChange={changeHandler} kompakt visLabel>
  <StegIndikator.Steg ferdig>Steg 1 label</StegIndikator.Steg>
  <StegIndikator.Steg ferdig aktivt >Steg 2 label</StegIndikator.Steg>
  <StegIndikator.Steg ferdig >Steg 3 label</StegIndikator.Steg>
</StegIndikator>
```
Evt. 
```
<StegIndikator 
  onChange={changeHandler} 
  kompakt 
  visLabel
  steg={[
    { label: 'Steg 1', ferdig: true },
    { label: 'Steg 2', ferdig: true, aktiv: true },
    { label: 'Steg 3', ferdig: true },
  ]}
/>
```

Disse to er heller ikke mutually exclusive på noen måte, og det er mulig å lage en komponent som enkelt støtter begge formatene. Hvis man i tillegg ønsker en komponent som er helt self-contained så ville jeg laget det som en ny komponent rundt en av disse, slik at vi får splittet ut logikken (Litt sånn som nav-frontend-ekspanderbartpanel).

Men kan være at det er noe funksjonalitet som jeg bare ikke har fått med meg i denne PR som ikke er støttet. Men hva tenker dere @Lillebo  @erlendev  @frodehansen2 ?

Reply:
Gir mye mening dette :+1: Tenkte kanskje det ville være enklere å bare sende inn et tall på parent for å angi aktivt steg, og så la komponenten få ansvaret med å delegere til riktig barn deretter. Kan gjerne flytte dette ned som props på stegene.

Litt mer usikker ang. `onChange`. Vil det ikke være fare for at komponenten "flikker" hvis bruker forsøker å navigere til steg som ikke er tillatt? Av UU-hensyn bør vi visstnok unngå bruk av `disabled` og heller gi feilmelding ved ulovlige klikk. Tenker du samtidig å heller bruke radio-knapper til dette?

Reply:
At `disabled` ikke kan brukes av UU-hensyn er nytt for meg. Har du noe mer info om dette? 
Om det skal brukes radio eller button er egentlig ikke så viktig for min del, så her ville jeg nok bare hørt med David hva han synes virker mest logisk.

Reply:
Er vel mer en anbefaling, fremfor forbud/krav. Husker ikke helt hvor jeg hørte det - tror det var @karinaldw som nevnte det for meg en gang vi diskuterte skjema-validering. Men fant denne i farta: https://axesslab.com/disabled-buttons-suck/

Skal høre med David 👍

Reply:
Tenker uansett at button/radios kan endres på senere, da dette blir en implementasjonsdetalj internt i komponenten. Mens apiet til komponenten er vanskeligere å endre på med en gang komponenten har blitt tatt ibruk.